File: framework/wcf/Extensibility/Transactions/TransactionMessagePropertyUdpTransport/VB/MessageInspectors/TransactionFlowBehavior.vb
'  Copyright (c) Microsoft Corporation. All rights reserved.

Imports System.Security.Permissions
Imports System.ServiceModel.Description

Namespace Microsoft.ServiceModel.Samples
	Public Class TransactionFlowBehavior
		Implements IEndpointBehavior
        Public Sub AddBindingParameters(ByVal endpoint As ServiceEndpoint,
                                        ByVal bindingParameters As System.ServiceModel.Channels.BindingParameterCollection) Implements IEndpointBehavior.AddBindingParameters
        End Sub

        Public Sub ApplyClientBehavior(ByVal endpoint As ServiceEndpoint,
                                       ByVal clientRuntime As System.ServiceModel.Dispatcher.ClientRuntime) Implements IEndpointBehavior.ApplyClientBehavior
            Dim inspector As New TransactionFlowInspector()
            clientRuntime.MessageInspectors.Add(inspector)
        End Sub

        Public Sub ApplyDispatchBehavior(ByVal endpoint As ServiceEndpoint,
                                         ByVal endpointDispatcher As System.ServiceModel.Dispatcher.EndpointDispatcher) Implements IEndpointBehavior.ApplyDispatchBehavior
        End Sub

		Public Sub Validate(ByVal endpoint As ServiceEndpoint) Implements IEndpointBehavior.Validate
		End Sub
	End Class
End Namespace

File: framework/wcf/Extensibility/Transactions/TransactionMessagePropertyUdpTransport/CS/MessageInspectors/TransactionFlowBehavior.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System.Security.Permissions;
using System.ServiceModel.Description;

namespace Microsoft.ServiceModel.Samples
{
    public class TransactionFlowBehavior : IEndpointBehavior
    {
        public void AddBindingParameters(ServiceEndpoint endpoint, System.ServiceModel.Channels.BindingParameterCollection bindingParameters)
        {
        }

        public void ApplyClientBehavior(ServiceEndpoint endpoint, System.ServiceModel.Dispatcher.ClientRuntime clientRuntime)
        {
            TransactionFlowInspector inspector = new TransactionFlowInspector();
            clientRuntime.MessageInspectors.Add(inspector);
        }

        public void ApplyDispatchBehavior(ServiceEndpoint endpoint, System.ServiceModel.Dispatcher.EndpointDispatcher endpointDispatcher)
        {
        }

        public void Validate(ServiceEndpoint endpoint)
        {
        }
    }
}
